### DISCUSSION TITLE
How can I interpolate missing y values in a line chart?

### DISCUSSION THREAD

**Discussion by emkayDauda** [patrykandpatrick/vico#965] [Questions] [answered]
### Question

I have a dataset with missing values, and I would like the line chart to interpolate the missing points so the line continues smoothly across gaps. For example, with the following data:

```
val sampleLineChartData = ChartData(
    seriesValues = listOf(80, 110, ?, ?, ?, 70, 85),
    bottomAxisValues = listOf("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul")
)
```

I want the line to connect the known values (Feb to Jun in this case), while keeping the missing months (e.g., Mar to May) as interpolated values. How can I achieve this in the chart library while keeping persistent markers only on actual data points?

### Vico version(s)

2.0.0-beta.2

### UI framework(s)

Jetpack Compose

**Comment by Gowsky** [comment, marked as answer]
Hello, @emkayDauda.

1. Add the _y_ values that are available and specify their _x_ values explicitly. You can do this via the `series` overload with `x` and `y` parameters. The entries will be positioned proportionally, as that's how the Cartesian coordinate system works.
2. Use the `getXStep` parameter of `rememberCartesianChart` to lock the _x_ step at one day. For the data you provided, it'll be one day by default, but you should lock it to ensure that it remains the same when, e.g., you have data only for every second day.

> while keeping persistent markers only on actual data points

When adding persistent markers, you specify their _x_ values yourself. Therefore, you should simply add persistent markers only for the _x_ values that have _y_ values (Also, persistent markers can be added only where there’s an entry, so even if you specified an _x_ value that doesn’t have an entry, a marker wouldn’t be displayed.)

However, if you need indication at every entry, then you shouldn't use persistent markers but rather points. See the `point` parameter of `LineCartesianLayer.rememberLine`. Points will automatically be added only for the data points, as desired.
